solr/core/src/java/org/apache/solr/core/SolrResourceLoader.java

```diff
@@ -694,9 +694,14 @@ public <T> boolean addToCoreAware(T obj) {
     }
   }
 
+  void initConfig(SolrConfig config) {
+    assert this.config == null || this.config == config;
```

Comment:
Does this assertion make sense?

I can see that SolrConfig has a `private final` reference to the resource loader, but is it OK to assume the opposite - that a resource loader is only ever associated with a single SolrConfig instance? Or is it an actual invariant that Solr enforces?

Reply:
Good question, and honestly when i see the words "resource loader" I run and hide!   Not familiar with that.  Maybe ask on the dev list?

Reply:
Based on the code observation, an instance of SolrConfig is always created via `SolrConfig#readFromResourceLoader` or the public "use-in-tests-only" ctor - and at the call sites, there's always a new instance of SolrResourceLoader being created.

So that indicates to me that the lifecycles of the config and the loader are naturally tied together.